HPXMLtoOpenStudio/resources/defaults.rb

```diff
@@ -5570,6 +5800,346 @@ def self.get_ceiling_fan_months(weather)
+            end
+          else # max
+            if heat_pump.backup_heating_fuel == HPXML::FuelTypeElectricity
+              watts += [HVAC.get_dx_heating_coil_power_watts_from_capacity(UnitConversions.convert(heat_pump.heating_capacity, 'btu/hr', 'kbtu/hr')),
+                        UnitConversions.convert(HVAC.get_heating_input_capacity(heat_pump.backup_heating_capacity, heat_pump.backup_heating_efficiency_afue, heat_pump.backup_heating_efficiency_percent), 'btu/hr', 'w')].max
+            else
+              watts += HVAC.get_dx_heating_coil_power_watts_from_capacity(UnitConversions.convert(heat_pump.heating_capacity, 'btu/hr', 'kbtu/hr'))
+            end
+          end
+
+          if heat_pump.backup_heating_fuel == HPXML::FuelTypeElectricity
+            breaker_spaces += get_breaker_spaces_from_backup_heating_capacity(heat_pump.backup_heating_capacity) # AHU
+          else
+            breaker_spaces += 1 # AHU
+          end
+        else # separate or none
+          watts += HVAC.get_dx_heating_coil_power_watts_from_capacity(UnitConversions.convert(heat_pump.heating_capacity, 'btu/hr', 'kbtu/hr'))
+        end
+
+        watts += HVAC.get_blower_fan_power_watts(heat_pump.fan_watts_per_cfm, heat_pump.heating_airflow_cfm)
+
+        breaker_spaces += 2 # ODU
+      end
+
+    elsif type == HPXML::ElectricPanelLoadTypeCooling
+      hpxml_bldg.cooling_systems.each do |cooling_system|
+        next if !system_ids.include?(cooling_system.id)
+        next if cooling_system.is_shared_system
+        next if cooling_system.fraction_cool_load_served == 0
+
+        watts += HVAC.get_dx_cooling_coil_power_watts_from_capacity(UnitConversions.convert(cooling_system.cooling_capacity, 'btu/hr', 'kbtu/hr'))
+        watts += HVAC.get_blower_fan_power_watts(cooling_system.fan_watts_per_cfm, cooling_system.cooling_airflow_cfm)
+
+        heating_system = cooling_system.attached_heating_system
+        if !heating_system.nil? &&
+           ((heating_system.is_a? HPXML::HeatingSystem) && (heating_system.heating_system_fuel != HPXML::FuelTypeElectricity))
+          breaker_spaces += 3 # AHU; paired w/fuel heating system
+        elsif voltage == 240
+          breaker_spaces += 2 # AHU
+        end
+      end
+
+      hpxml_bldg.heat_pumps.each do |heat_pump|
+        next if !system_ids.include?(heat_pump.id)
+        next if heat_pump.fraction_cool_load_served == 0
+
+        watts += HVAC.get_dx_cooling_coil_power_watts_from_capacity(UnitConversions.convert(heat_pump.cooling_capacity, 'btu/hr', 'kbtu/hr'))
+        watts += HVAC.get_blower_fan_power_watts(heat_pump.fan_watts_per_cfm, heat_pump.cooling_airflow_cfm)
+
+        if heat_pump.fraction_heat_load_served == 0
+          breaker_spaces += 3 # ODU; the 3 we missed adding to heating
+        end
+      end
+
+    elsif type == HPXML::ElectricPanelLoadTypeWaterHeater
```

Comment:
Can we move these assignments to a table?

Reply:
Which of the following were you thinking of for a table? All of these?
- ElectricPanelLoadTypeWaterHeater
- ElectricPanelLoadTypeClothesDryer
- ElectricPanelLoadTypeDishwasher
- ElectricPanelLoadTypeRangeOven
- ElectricPanelLoadTypeMechVent
- ElectricPanelLoadTypePermanentSpaPump
- ElectricPanelLoadTypePermanentSpaHeater
- ElectricPanelLoadTypePoolHeater
- ElectricPanelLoadTypePoolPump
- ElectricPanelLoadTypeWellPump
- ElectricPanelLoadTypeElectricVehicleCharging
- ElectricPanelLoadTypeLighting (i.e., the floor area multiplier?)
- ElectricPanelLoadTypeKitchen
- ElectricPanelLoadTypeLaundry
- ElectricPanelLoadTypeOther

What would be the easiest format to read in, considering some of the apply logic needed?

Reply:
Are you using "apply logic" to mean that basically each of the above categories is somewhat unique in that power ratings are assigned based on different properties? For example, water heater uses number of bathrooms but well pump uses number of bedrooms?

So a flat table would be pretty sparse. Yeah, I'm not really sure what the best format is. Maybe table (or similar) isn't the best approach -- I was just trying to think of a way these power rating values could be easily updated/maintained without having to read through a bunch of lines of code.